pandas/core/groupby/groupby.py

```diff
@@ -2298,7 +2298,28 @@ def levels(self):
 
     @property
     def names(self):
-        return [ping.name for ping in self.groupings]
+        # add suffix to level name in case they contain duplicates (GH 19029):
```

Comment:
this is super complicated, what exactly are you trying to do here?

Reply:
Hi. The goal here is to add a suffix to duplicate entries:
```python
# takes care of multiplicity:
['x', 'x', 'y', 'y']  => ['x_0', 'x_1', 'y_0', 'y_1']
```
Before the current version the code just enumerated all the entries regardless of their multiplicity:
```python
['x', 'x', 'y', 'y']  => ['x_0', 'x_1', 'y_2', 'y_3']
```
For some reason I thought that the current version would be better suited.
I can switch back to the old (and probably less confusing) version if you want.